pygama/lgdo/vectorofvectors.py

```diff
@@ -47,7 +47,9 @@ def __init__(self, flattened_data=None, cumulative_length=None, shape_guess=None
             length = np.prod(shape_guess)
             self.flattened_data = Array(shape=(length,), dtype=dtype)
         else: self.flattened_data = flattened_data
-        self.dtype = self.flattened_data.dtype
+        if dtype is None:
+            self.dtype = self.flattened_data.dtype
+        else: self.dtype = dtype
```

Comment:
Oh this is an interesting block. There are 4 options:
- flattened_data is None and dtype is not None: this is supposed to not be allowed; presumably Array's __init__ function should emit a warning in this case, but it looks like it doesn't. I think this was already a problem in the original code that should probably be fixed.
- flattened_data is None and dtype is not None: make an empty array with the requested dtype (ok in both versions)
- flattened data is not None and dtype is None: use the input array and its keep its dtype. This is correct in both versions.
- flattened data is not None and dtype is not None: use the input array but change its dtype if necessary. This appears to not be implemented in this way in both the new and the old code.